### ISSUE TITLE
Reconciliation failure, pod in CrashLoopBackoff

### ISSUE THREAD

**Issue by ianpartridge** [application-stacks/runtime-component-operator#93]
### What did you do?

`appsody init quarkus && appsody deploy`.

### What did you expect to see?

Operator up and running. Quarkus deployed.

### What did you see instead?

```
$ kubectl get all
NAME                                    READY   STATUS             RESTARTS   AGE
pod/appsody-operator-5699df549f-6kcrl   0/1     CrashLoopBackOff   4          3m5s
pod/quarkus-test-7b7848c7df-2pldk       1/1     Running            0          3m1s


NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
service/kubernetes     ClusterIP   10.96.0.1       <none>        443/TCP          10d
service/quarkus-test   NodePort    10.108.11.197   <none>        8080:32256/TCP   6m31s


NAME                               READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/appsody-operator   0/1     1            0           3m5s
deployment.apps/quarkus-test       1/1     1            1           3m1s

NAME                                          DESIRED   CURRENT   READY   AGE
replicaset.apps/appsody-operator-5699df549f   1         1         0       3m5s
replicaset.apps/quarkus-test-7b7848c7df       1         1         1       3m1s
```

```
$ kubectl logs pod/appsody-operator-5699df549f-j565v
{"level":"info","ts":1588071457.4020913,"logger":"cmd","msg":"Operator Version: 0.5.0"}
{"level":"info","ts":1588071457.4022222,"logger":"cmd","msg":"Go Version: go1.13.10"}
{"level":"info","ts":1588071457.4022381,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1588071457.4022946,"logger":"cmd","msg":"Version of operator-sdk: v0.15.2"}
{"level":"info","ts":1588071457.4028146,"logger":"leader","msg":"Trying to become the leader."}
{"level":"info","ts":1588071457.8237953,"logger":"leader","msg":"Found existing lock with my name. I was likely restarted."}
{"level":"info","ts":1588071457.8242588,"logger":"leader","msg":"Continuing as the leader."}
{"level":"info","ts":1588071458.2282734,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":"0.0.0.0:8383"}
{"level":"info","ts":1588071458.2287433,"logger":"cmd","msg":"Registering Components."}
{"level":"info","ts":1588071458.229575,"logger":"controller_appsodyapplication","msg":"newReconciler","watchNamespaces":["default"]}
{"level":"info","ts":1588071458.257238,"logger":"cmd","msg":"Starting the Cmd."}
{"level":"info","ts":1588071458.2578323,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
{"level":"info","ts":1588071458.3583388,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588071458.358462,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588071458.459391,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588071458.5600038,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588071458.6633706,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588071458.7640672,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588071458.865369,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588071458.865641,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588071458.8658004,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"appsodyapplication-controller"}
{"level":"info","ts":1588071458.8658307,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"appsodyapplication-controller","worker count":1}
{"level":"info","ts":1588071458.8660092,"logger":"controller_appsodyapplication","msg":"Reconciling AppsodyApplication","Request.Namespace":"default","Request.Name":"quarkus-test"}
{"level":"info","ts":1588071459.102293,"logger":"utils","msg":"Reconciled","Kind":"ServiceAccount","Name":"quarkus-test","Status":"unchanged"}
{"level":"info","ts":1588071459.104488,"logger":"utils","msg":"Reconciled","Kind":"Service","Name":"quarkus-test","Status":"unchanged"}
{"level":"info","ts":1588071459.1230283,"logger":"utils","msg":"Reconciled","Kind":"Deployment","Name":"quarkus-test","Status":"updated"}
E0428 10:57:39.233664       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 453 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x15c8800, 0x25b3bf0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x15c8800, 0x25b3bf0)
	/home/travis/.gimme/versions/go1.13.10.linux.amd64/src/runtime/panic.go:679 +0x1b2
github.com/application-stacks/runtime-component-operator/pkg/utils.CustomizeIngress(0xc00087c600, 0x1a2d8c0, 0xc00098a600)
	appsody-operator/vendor/github.com/application-stacks/runtime-component-operator/pkg/utils/utils.go:927 +0x104
github.com/appsody/appsody-operator/pkg/controller/appsodyapplication.(*ReconcileAppsodyApplication).Reconcile.func10(0x0, 0x0)
	appsody-operator/pkg/controller/appsodyapplication/appsodyapplication_controller.go:782 +0x3f
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate(0xc0005e7688, 0xc000ab2479, 0x7, 0xc000ab2460, 0xc, 0x19c7020, 0xc00087c600, 0x19c7020, 0xc00087c600)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil/controllerutil.go:172 +0x2b
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate(0x19f4660, 0xc0000b0008, 0x1a0ec00, 0xc000760150, 0x19c7020, 0xc00087c600, 0xc0005e7688, 0x40bdc6, 0xc00087c600, 0x180, ...)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil/controllerutil.go:146 +0x1a0
github.com/application-stacks/runtime-component-operator/pkg/utils.(*ReconcilerBase).CreateOrUpdate(0xc00012e850, 0x1a2f960, 0xc00087c600, 0x1a2e880, 0xc00098a600, 0xc0005e7688, 0x0, 0x0)
	appsody-operator/vendor/github.com/application-stacks/runtime-component-operator/pkg/utils/reconciler.go:93 +0x1d3
github.com/appsody/appsody-operator/pkg/controller/appsodyapplication.(*ReconcileAppsodyApplication).Reconcile(0xc00012e850, 0xc00082e549, 0x7, 0xc00082e530, 0xc, 0xc000949cd8, 0xc000838900, 0xc0001c8098, 0x19caa40)
	appsody-operator/pkg/controller/appsodyapplication/appsodyapplication_controller.go:781 +0x2e78
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0008ca0c0, 0x16223c0, 0xc00041e0a0, 0xc000219300)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0008ca0c0, 0xc00075ee00)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0008ca0c0)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00009f9c0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00009f9c0, 0x3b9aca00, 0x0, 0x1, 0xc00009dda0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc00009f9c0, 0x3b9aca00, 0xc00009dda0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193 +0x328
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x132e5a4]

goroutine 453 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x15c8800, 0x25b3bf0)
	/home/travis/.gimme/versions/go1.13.10.linux.amd64/src/runtime/panic.go:679 +0x1b2
github.com/application-stacks/runtime-component-operator/pkg/utils.CustomizeIngress(0xc00087c600, 0x1a2d8c0, 0xc00098a600)
	appsody-operator/vendor/github.com/application-stacks/runtime-component-operator/pkg/utils/utils.go:927 +0x104
github.com/appsody/appsody-operator/pkg/controller/appsodyapplication.(*ReconcileAppsodyApplication).Reconcile.func10(0x0, 0x0)
	appsody-operator/pkg/controller/appsodyapplication/appsodyapplication_controller.go:782 +0x3f
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate(0xc0005e7688, 0xc000ab2479, 0x7, 0xc000ab2460, 0xc, 0x19c7020, 0xc00087c600, 0x19c7020, 0xc00087c600)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil/controllerutil.go:172 +0x2b
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate(0x19f4660, 0xc0000b0008, 0x1a0ec00, 0xc000760150, 0x19c7020, 0xc00087c600, 0xc0005e7688, 0x40bdc6, 0xc00087c600, 0x180, ...)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil/controllerutil.go:146 +0x1a0
github.com/application-stacks/runtime-component-operator/pkg/utils.(*ReconcilerBase).CreateOrUpdate(0xc00012e850, 0x1a2f960, 0xc00087c600, 0x1a2e880, 0xc00098a600, 0xc0005e7688, 0x0, 0x0)
	appsody-operator/vendor/github.com/application-stacks/runtime-component-operator/pkg/utils/reconciler.go:93 +0x1d3
github.com/appsody/appsody-operator/pkg/controller/appsodyapplication.(*ReconcileAppsodyApplication).Reconcile(0xc00012e850, 0xc00082e549, 0x7, 0xc00082e530, 0xc, 0xc000949cd8, 0xc000838900, 0xc0001c8098, 0x19caa40)
	appsody-operator/pkg/controller/appsodyapplication/appsodyapplication_controller.go:781 +0x2e78
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0008ca0c0, 0x16223c0, 0xc00041e0a0, 0xc000219300)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0008ca0c0, 0xc00075ee00)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0008ca0c0)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00009f9c0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00009f9c0, 0x3b9aca00, 0x0, 0x1, 0xc00009dda0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc00009f9c0, 0x3b9aca00, 0xc00009dda0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193 +0x328
```

### Environment

macOS 10.15.4 running Docker Desktop with Kubernetes:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by ianpartridge** [comment]
Hmm, I have now got this to deploy after deleting everything (including the `AppsodyApplication` CRD).. not quite sure what happened but it was definitely broken..

**Comment by ianpartridge** [comment]
Spoke too soon... the app successfully deployed but the operator was sick afterwards...

```
$ kubectl get all
NAME                                    READY   STATUS    RESTARTS   AGE
pod/appsody-operator-5699df549f-9j94f   0/1     Error     3          4m33s
pod/quarkus-test-78b8697757-gcxb4       1/1     Running   0          69s


NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
service/kubernetes     ClusterIP   10.96.0.1       <none>        443/TCP          10d
service/quarkus-test   NodePort    10.98.216.198   <none>        8080:31496/TCP   69s


NAME                               READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/appsody-operator   0/1     1            0           4m33s
deployment.apps/quarkus-test       1/1     1            1           69s

NAME                                          DESIRED   CURRENT   READY   AGE
replicaset.apps/appsody-operator-5699df549f   1         1         0       4m33s
replicaset.apps/quarkus-test-78b8697757       1         1         1       69s

$ kubectl logs pod/appsody-operator-5699df549f-9j94f
{"level":"info","ts":1588076271.9951715,"logger":"cmd","msg":"Operator Version: 0.5.0"}
{"level":"info","ts":1588076271.9952302,"logger":"cmd","msg":"Go Version: go1.13.10"}
{"level":"info","ts":1588076271.9952435,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1588076271.9952695,"logger":"cmd","msg":"Version of operator-sdk: v0.15.2"}
{"level":"info","ts":1588076271.995498,"logger":"leader","msg":"Trying to become the leader."}
{"level":"info","ts":1588076272.4159346,"logger":"leader","msg":"Found existing lock with my name. I was likely restarted."}
{"level":"info","ts":1588076272.4160545,"logger":"leader","msg":"Continuing as the leader."}
{"level":"info","ts":1588076272.8195276,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":"0.0.0.0:8383"}
{"level":"info","ts":1588076272.820238,"logger":"cmd","msg":"Registering Components."}
{"level":"info","ts":1588076272.8216546,"logger":"controller_appsodyapplication","msg":"newReconciler","watchNamespaces":["default"]}
{"level":"info","ts":1588076272.85471,"logger":"cmd","msg":"Starting the Cmd."}
{"level":"info","ts":1588076272.8554337,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
{"level":"info","ts":1588076272.9564567,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588076272.9566088,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588076273.0577385,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588076273.1588142,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588076273.2598634,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588076273.3611765,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588076273.4617906,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588076273.4619253,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"appsodyapplication-controller","source":"kind source: /, Kind="}
{"level":"info","ts":1588076273.4624677,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"appsodyapplication-controller"}
{"level":"info","ts":1588076273.4625416,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"appsodyapplication-controller","worker count":1}
{"level":"info","ts":1588076273.462771,"logger":"controller_appsodyapplication","msg":"Reconciling AppsodyApplication","Request.Namespace":"default","Request.Name":"quarkus-test"}
{"level":"info","ts":1588076273.704733,"logger":"utils","msg":"Reconciled","Kind":"ServiceAccount","Name":"quarkus-test","Status":"unchanged"}
{"level":"info","ts":1588076273.7062447,"logger":"utils","msg":"Reconciled","Kind":"Service","Name":"quarkus-test","Status":"unchanged"}
{"level":"info","ts":1588076273.7228706,"logger":"utils","msg":"Reconciled","Kind":"Deployment","Name":"quarkus-test","Status":"updated"}
E0428 12:17:53.828737       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 413 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x15c8800, 0x25b3bf0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x15c8800, 0x25b3bf0)
	/home/travis/.gimme/versions/go1.13.10.linux.amd64/src/runtime/panic.go:679 +0x1b2
github.com/application-stacks/runtime-component-operator/pkg/utils.CustomizeIngress(0xc0004ed200, 0x1a2d8c0, 0xc0005a4f00)
	appsody-operator/vendor/github.com/application-stacks/runtime-component-operator/pkg/utils/utils.go:927 +0x104
github.com/appsody/appsody-operator/pkg/controller/appsodyapplication.(*ReconcileAppsodyApplication).Reconcile.func10(0x0, 0x0)
	appsody-operator/pkg/controller/appsodyapplication/appsodyapplication_controller.go:782 +0x3f
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate(0xc000a65688, 0xc000a38c99, 0x7, 0xc000a38c80, 0xc, 0x19c7020, 0xc0004ed200, 0x19c7020, 0xc0004ed200)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil/controllerutil.go:172 +0x2b
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate(0x19f4660, 0xc0000b0008, 0x1a0ec00, 0xc0007cc210, 0x19c7020, 0xc0004ed200, 0xc000a65688, 0x40bdc6, 0xc0004ed200, 0x180, ...)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil/controllerutil.go:146 +0x1a0
github.com/application-stacks/runtime-component-operator/pkg/utils.(*ReconcilerBase).CreateOrUpdate(0xc0001479d0, 0x1a2f960, 0xc0004ed200, 0x1a2e880, 0xc0005a4f00, 0xc000a65688, 0x0, 0x0)
	appsody-operator/vendor/github.com/application-stacks/runtime-component-operator/pkg/utils/reconciler.go:93 +0x1d3
github.com/appsody/appsody-operator/pkg/controller/appsodyapplication.(*ReconcileAppsodyApplication).Reconcile(0xc0001479d0, 0xc0007c3589, 0x7, 0xc0007c3570, 0xc, 0xc0004e6cd8, 0xc000310510, 0xc00050e1b8, 0x19caa40)
	appsody-operator/pkg/controller/appsodyapplication/appsodyapplication_controller.go:781 +0x2e78
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0006de0c0, 0x16223c0, 0xc0007d8840, 0x0)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0006de0c0, 0x1191100)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0006de0c0)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0005d6ed0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005d6ed0, 0x3b9aca00, 0x0, 0xc0001dce01, 0xc0002e7080)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0005d6ed0, 0x3b9aca00, 0xc0002e7080)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193 +0x328
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x132e5a4]

goroutine 413 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x15c8800, 0x25b3bf0)
	/home/travis/.gimme/versions/go1.13.10.linux.amd64/src/runtime/panic.go:679 +0x1b2
github.com/application-stacks/runtime-component-operator/pkg/utils.CustomizeIngress(0xc0004ed200, 0x1a2d8c0, 0xc0005a4f00)
	appsody-operator/vendor/github.com/application-stacks/runtime-component-operator/pkg/utils/utils.go:927 +0x104
github.com/appsody/appsody-operator/pkg/controller/appsodyapplication.(*ReconcileAppsodyApplication).Reconcile.func10(0x0, 0x0)
	appsody-operator/pkg/controller/appsodyapplication/appsodyapplication_controller.go:782 +0x3f
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate(0xc000a65688, 0xc000a38c99, 0x7, 0xc000a38c80, 0xc, 0x19c7020, 0xc0004ed200, 0x19c7020, 0xc0004ed200)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil/controllerutil.go:172 +0x2b
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate(0x19f4660, 0xc0000b0008, 0x1a0ec00, 0xc0007cc210, 0x19c7020, 0xc0004ed200, 0xc000a65688, 0x40bdc6, 0xc0004ed200, 0x180, ...)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil/controllerutil.go:146 +0x1a0
github.com/application-stacks/runtime-component-operator/pkg/utils.(*ReconcilerBase).CreateOrUpdate(0xc0001479d0, 0x1a2f960, 0xc0004ed200, 0x1a2e880, 0xc0005a4f00, 0xc000a65688, 0x0, 0x0)
	appsody-operator/vendor/github.com/application-stacks/runtime-component-operator/pkg/utils/reconciler.go:93 +0x1d3
github.com/appsody/appsody-operator/pkg/controller/appsodyapplication.(*ReconcileAppsodyApplication).Reconcile(0xc0001479d0, 0xc0007c3589, 0x7, 0xc0007c3570, 0xc, 0xc0004e6cd8, 0xc000310510, 0xc00050e1b8, 0x19caa40)
	appsody-operator/pkg/controller/appsodyapplication/appsodyapplication_controller.go:781 +0x2e78
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0006de0c0, 0x16223c0, 0xc0007d8840, 0x0)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0006de0c0, 0x1191100)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0006de0c0)
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0005d6ed0)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005d6ed0, 0x3b9aca00, 0x0, 0xc0001dce01, 0xc0002e7080)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0005d6ed0, 0x3b9aca00, 0xc0002e7080)
	appsody-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	appsody-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:193 +0x328
```

**Comment by arturdzm** [comment]
This might be due to `route` field not being set (nil).
Try setting `route :{}`

**Comment by ianpartridge** [comment]
Thanks for the quick reply :)

Sorry, but how do I set route?

**Comment by arturdzm** [comment]
Just add  in your  app CR

I think the issue is that when Ingress is enabled, it tries to read the route field. 

```
Kind: AppsodyApplication
spec:
  expose: true
  route: {}
```
I think this could be a bug.
it should not be required if the `defaultHostname` is specified in operator config map.
Or user needs to provide `spec.route.host` field.
In any case the operator should not crash

**Comment by ianpartridge** [comment]
I confirm that if I edit my `app-deploy.yaml` and set `spec.route` to `{}` then the operator does not crash.

**Comment by arturdzm** [comment]
Fixed in 0.5.1 release. Available now on Docker Hub and soon should be on operator hub as well

**Comment by ianpartridge** [comment]
Thank you very much for the quick fix!
